ThreeBotPackages/zerobot/webinterface/chatflows/components/NodeSelector.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+    <div>
+      <Message :payload="payload"></Message>
+      <v-autocomplete v-model="val" :hint="hint" :rules="rules" :persistent-hint="true" label="Nodes" :items="nodes" validate-on-blur outlined small-chips :multiple="multiple"></v-autocomplete>
+      
+      <h3 class="title font-regular mb-3 mt-5">Filter Nodes</h3>
+      <v-autocomplete v-model="filters.farm" label="Farm" :items="farms" validate-on-blur outlined small-chips></v-autocomplete>
+      <v-slider class="mt-5" label="CRU" v-model="filters.cru" max="64" thumb-label="always"></v-slider>
+      <v-slider class="mt-3" label="MRU" v-model="filters.mru" max="512" thumb-label="always"></v-slider>
+      <v-slider class="mt-3" label="HRU" v-model="filters.hru" max="5000" thumb-label="always"></v-slider>
+      <v-slider class="mt-3" label="SRU" v-model="filters.sru" max="5000" thumb-label="always"></v-slider>
+    </div>
+</template>
+
+<script>
+  const axios = require('axios')
+  const baseUrl = "/zerobot/webinterface/actors"
+  
+  module.exports = {
+    mixins: [field],
+    props: {payload: Object},
+    data () {
+      return {
+        filters: {
+          cru: null,
+          mru: null,
+          hru: null,
+          sru: null,
+          country: null,
+          farm: null
+        },
+        nodes: [],
+        farms: []
+      }
+    },
+    watch: {
+      filters: {
+        handler () {
+          this.findNodes()
+        },
+        deep: true
+      }
+    },
+    computed: {
+      hint () {
+        if (this.multiple) {
+          return `${this.nodes.length} Nodes, ${this.val.length} Selected`
+        } else {
+          return `${this.nodes.length} Nodes, ${this.val.length ? 1 : 0} Selected`
+        }
+      },
+      multiple () {
+        return this.payload.kwargs.multiple === true
+      }
+    },
+    methods: {
+      findNodes () {
```

Comment:
This function is too naive. What you are querying is the total amount of resources. While what you are interested in is the amount of capacity still free on the node. 
You need to do some math: `total - used = free`

You should also most probably filter out any offline nodes.
There is also another think we identified recently regarding the health of the node. It is the proper network configuration of a node. A node that has an unhealthy network config should also be skipped cause it is not able to process reservations. See https://github.com/threefoldtech/jumpscaleX_threebot/pull/690 for more detail

Reply:
I guess some extra logic so you do keep the nodes already loaded in cache would also allow to have less requests going to the explorer. Now any time you modify a slider you send a new request to the explorer. This is going to make a lot of requests...